src/webpack.ts

```diff
@@ -0,0 +1,118 @@
+import { BuildContext, TaskInfo } from './util/interfaces';
+import { BuildError, Logger } from './util/logger';
+import { cacheTranspiledTsFiles, setModulePathsCache } from './util/helpers';
+import { fillConfigDefaults, generateContext, getUserConfigFile, replacePathVars } from './util/config';
+import { join } from 'path';
+import * as wp from 'webpack';
+
+
+export function webpack(context: BuildContext, configFile: string) {
+  context = generateContext(context);
+  configFile = getUserConfigFile(context, taskInfo, configFile);
+
+  cacheTranspiledTsFiles(context.tsFiles);
+
+  const logger = new Logger('webpack');
+
+  return webpackWorker(context, configFile)
+    .then(() => {
+      logger.finish();
+    })
+    .catch(err => {
+      throw logger.fail(err);
+    });
+}
+
+
+export function webpackUpdate(event: string, path: string, context: BuildContext, configFile: string) {
+  configFile = getUserConfigFile(context, taskInfo, configFile);
+  const logger = new Logger('webpack update');
+
+  cacheTranspiledTsFiles(context.tsFiles);
+
+  return webpackWorker(context, configFile)
+    .then(() => {
+      logger.finish();
+    })
+    .catch(err => {
+      throw logger.fail(err);
+    });
+}
+
+
+export function webpackWorker(context: BuildContext, configFile: string): Promise<any> {
+  return new Promise((resolve, reject) => {
+    try {
+      const webpackConfig = getWebpackConfig(context, configFile);
+      const compiler = wp(webpackConfig);
+
+      compiler.run((err: Error, stats: any) => {
+        if (err) {
+          reject(err);
+
+        } else {
+          // set the module files used in this bundle
+          // this reference can be used elsewhere in the build (sass)
+          const files = stats.compilation.modules.map((webpackObj: any) => {
```

Comment:
@TheLarkInn, any idea why `resource` might be null on some modules when I loop over them? I am filtering it out or just using context for my purposes, but it was surprising to have `resources` populated correctly 999/1000ish times.

Thanks,
Dan